library/standard_tables_capture.inc.php

```diff
@@ -502,142 +502,90 @@ function icd_import($type)
-        '#FLD4#' => "ICD9_02", '#POS4#' => 17, '#LEN4#' => 5,
-        '#FLD5#' => "ICD9_03", '#POS5#' => 23, '#LEN5#' => 5,
-        '#FLD6#' => "ICD9_04", '#POS6#' => 29, '#LEN6#' => 5,
-        '#FLD7#' => "ICD9_05", '#POS7#' => 35, '#LEN7#' => 5,
-        '#FLD8#' => "ICD9_06", '#POS8#' => 41, '#LEN8#' => 5);
-        $incoming['reimb_map_dx_'] = array('#TABLENAME#' => "icd10_reimbr_dx_9_10",
-        '#FLD1#' => "code", '#POS1#' => 1, '#LEN1#' => 7,
-        '#FLD2#' => "code_cnt", '#POS2#' => 9, '#LEN2#' => 1,
-        '#FLD3#' => "ICD9_01", '#POS3#' => 11, '#LEN3#' => 5,
-        '#FLD4#' => "ICD9_02", '#POS4#' => 17, '#LEN4#' => 5,
-        '#FLD5#' => "ICD9_03", '#POS5#' => 23, '#LEN5#' => 5,
-        '#FLD6#' => "ICD9_04", '#POS6#' => 29, '#LEN6#' => 5,
-        '#FLD7#' => "ICD9_05", '#POS7#' => 35, '#LEN7#' => 5,
-        '#FLD8#' => "ICD9_06", '#POS8#' => 41, '#LEN8#' => 5);
-        $incoming['I10gem'] = array('#TABLENAME#' => "icd10_gem_dx_10_9",
-        '#FLD1#' => "dx_icd10_source", '#POS1#' => 1, '#LEN1#' => 7,
-        '#FLD2#' => "dx_icd9_target", '#POS2#' => 9, '#LEN2#' => 5,
-        '#FLD3#' => "flags", '#POS3#' => 15, '#LEN3#' => 5);
-        $incoming['I9gem'] = array('#TABLENAME#' => "icd10_gem_dx_9_10",
-        '#FLD1#' => "dx_icd9_source", '#POS1#' => 1, '#LEN1#' => 5,
-        '#FLD2#' => "dx_icd10_target", '#POS2#' => 7, '#LEN2#' => 7,
-        '#FLD3#' => "flags", '#POS3#' => 15, '#LEN3#' => 5);
-        $incoming['gem_pcsi9'] = array('#TABLENAME#' => "icd10_gem_pcs_10_9",
-        '#FLD1#' => "pcs_icd10_source", '#POS1#' => 1, '#LEN1#' => 7,
-        '#FLD2#' => "pcs_icd9_target", '#POS2#' => 9, '#LEN2#' => 5,
-        '#FLD3#' => "flags", '#POS3#' => 15, '#LEN3#' => 5);
-        $incoming['gem_i9pcs'] = array('#TABLENAME#' => "icd10_gem_pcs_9_10",
-        '#FLD1#' => "pcs_icd9_source", '#POS1#' => 1, '#LEN1#' => 5,
-        '#FLD2#' => "pcs_icd10_target", '#POS2#' => 7, '#LEN2#' => 7,
-        '#FLD3#' => "flags", '#POS3#' => 15, '#LEN3#' => 5);
-    }
-
-    // set up the start of the load script to be appended from the incoming array defined above where incoming
-    // file matches
-    $db_load = "LOAD DATA LOCAL INFILE '#INFILE#' INTO TABLE #TABLENAME# FIELDS TERMINATED BY '\0' (@var) SET revision = 0, ";
-    $col_template = "#FLD# = trim(Substring(@var, #POS#, #LEN#))";
-
-    // load all data and set active revision
+
+    // find active revision
+    $res = sqlStatementNoLog("SELECT max(revision) rev FROM icd10_pcs_order_code");
+    $row = sqlFetchArray($res);
+    $next_rev = $row['rev'] + 1;
+    $incoming['icd10pcs_codes_'] = array(
+        'TABLENAME' => "icd10_pcs_order_code",
+        'FLD1' => "pcs_code", 'POS1' => 0, 'LEN1' => 7,
+        'FLD2' => "long_desc", 'POS2' => 8, 'LEN2' => 300,
+        'REV' => $next_rev
+    );
+
+    $res = sqlStatementNoLog("SELECT max(revision) rev FROM icd10_dx_order_code");
+    $row = sqlFetchArray($res);
+    $next_rev = $row['rev'] + 1;
```

Comment:
could use sqlQueryNoLog here instead since it just returns one row already in array form:
```
$rev = sqlQueryNoLog("SELECT max(revision) rev FROM icd10_dx_order_code")['rev'] ?? 0;
$next_rev = $rev + 1;
```

Reply:
also note above uses will deal ok when there is nothing in icd10_dx_order_code (ie. first time uploading codes)

Reply:
nice!

Reply:
guessing will have issues since variables don't match up. Would make the following changes:
```
$res = sqlQueryNoLog("SELECT max(revision) rev FROM icd10_dx_order_code");
$next_rev = ($res['rev'] ?? 0) + 1;
```

Reply:
thanks for catching that!